### PR TITLE
Improvements to dependencies in Makefile

### PR DESCRIPTION
**Pull request by brandoncurtis**
In the final example Makefile:

1) includes the Makefile configuration file, config.mk, in the project archive
2) adds additional dependencies to ensure that the archive is updated when analysis files are updated

### CODE REVIEW

File: code/09-conclusion-challenge-2/Makefile [hunk 1]
```diff
@@ -10,7 +10,7 @@ ZIPF_ARCHIVE=$(ZIPF_DIR).tar.gz
 .PHONY : all
 all : $(ZIPF_ARCHIVE)
 
-$(ZIPF_ARCHIVE) : $(ZIPF_DIR)
+$(ZIPF_ARCHIVE) : $(ZIPF_DIR) $(PNG_FILES) results.txt
```
Comment (gcapes) [inline code/09-conclusion-challenge-2/Makefile]: Hi Brandon,
Thanks for spotting this oversight. The behaviour should change, but I don't think your solution is the best way to do it. The dependencies are correct as they are - it's just that the `zipf_analysis` folder time stamp doesn't change when the updated files are copied into it. Hence it only works correctly the first time it is run. I propose leaving the dependencies as they are, but adding `touch $@` to the `$(ZIPF_DIR)` rule to update the time stamp.

```
$(ZIPF_DIR): Makefile results.txt \
             $(DAT_FILES) $(PNG_FILES) \
             $(COUNT_SRC) $(PLOT_SRC) $(ZIPF_SRC)
	mkdir -p $@
	cp $^ $@
	touch $@
```
Thanks

Comment (brandoncurtis) [inline code/09-conclusion-challenge-2/Makefile]: Thanks @gcapes!  This is definitely the better way to do it, so I've made this change.

**Comment by brandoncurtis** [comment]
I made the requested change and added a Challenge to the Episode that addresses why it's needed.

File: _episodes/09-conclusion.md [hunk 2]
```diff
@@ -98,3 +98,15 @@ The following figure shows the dependencies involved in building the `all` targe
...
+
+> ## Archiving the Makefile
+>
+> Why does the Makefile rule for the archive directory `touch` the archive directory after moving our code, data, plots and summary table into it?
+>
+> > ## Solution
+> > A directory's timestamp is not automatically updated when files are copied into it.
+> > If the code, data, plots, and summary table are updated and copied into the
+> > archive directory, the archive directory's timestamp must be updated with `touch`
+> > so that the target that makes `zipf_analysis.tar.gz` knows to run again.
```
Comment (gcapes) [inline _episodes/09-conclusion.md]: Please fix typo `target` to `rule`

> so that the target that makes ...


> so that the rule that makes ...

Comment (brandoncurtis) [inline _episodes/09-conclusion.md]: fixed!

**Comment by gcapes** [comment]
Thanks for your work!

**Pull request merged**